### PR TITLE
Fyle 85zrn57ak Code Coverage:Mileage rate services

### PR DESCRIPTION
**Pull request by CODEX108**

### CODE REVIEW

**Review by jayfyle — APPROVED** [review]

File: src/app/core/mock-data/mileage-rate.data.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { PlatformMileageRates } from '../models/platform/platform-mileage-rates.model';
 
-export const mileageRatesData: PlatformMileageRates[] = [
+export const unfilterdMileageRatesData: PlatformMileageRates[] = [
```
Comment (Dimple16) [inline src/app/core/mock-data/mileage-rate.data.ts:3]: `unfilteredMileageRatesData`

File: src/app/core/mock-data/mileage-rate.data.ts [hunk 2]
```diff
@@ -121,3 +121,201 @@ export const filterEnabledMileageRatesData: PlatformMileageRates[] = [
     readableRate: '₹122.00/mile',
   },
 ];
+
+export const nullRateIncludedData: PlatformMileageRates[] = [
```
Comment (Dimple16) [inline src/app/core/mock-data/mileage-rate.data.ts:125]: `nullMileageRateData`

File: src/app/core/mock-data/mileage-rate.data.ts [hunk 3]
```diff
@@ -121,3 +121,201 @@ export const filterEnabledMileageRatesData: PlatformMileageRates[] = [
...
+      },
+    ],
+    unit: 'MILES',
+    updated_at: new Date('2022-09-22T05:26:02.014758+00:00'),
+    vehicle_type: 'electric_car',
+    readableRate: '₹122.00/mile',
+  },
+];
+
+export const nullRateExcludedData1: PlatformMileageRates[] = [
```
Comment (Dimple16) [inline src/app/core/mock-data/mileage-rate.data.ts:177]: if it's excluded, then we can name it as `platformMileageRatesData`

File: src/app/core/services/mileage-rates.service.spec.ts [hunk 4]
```diff
@@ -20,24 +32,129 @@ describe('MileageRatesService', () => {
...
 
+  describe('formatMileageRateName()', () => {
+    it('should give the formated mileage rate name', () => {
+      const input = 'four_wheeler';
+      const output = mileageRatesService.formatMileageRateName(input);
+      expect(output).toEqual('Four Wheeler - Type 1');
+    });
+
+    it('should return the input string if it is not a valid rate name', () => {
+      const input = 'invalid_rate_name';
```
Comment (Dimple16) [inline src/app/core/services/mileage-rates.service.spec.ts:59]: `three_wheeler`

File: src/app/core/services/mileage-rates.service.spec.ts [hunk 5]
```diff
@@ -20,24 +32,129 @@ describe('MileageRatesService', () => {
...
+      currencyPipe.transform.and.returnValue('$10.50');
+      const result = mileageRatesService.getReadableRate(rate, currency, unit);
+      expect(currencyPipe.transform).toHaveBeenCalledOnceWith(rate, currency, 'symbol', '1.2-2');
+      expect(result).toEqual(expectedResult);
+    });
+  });
+
   it('filterEnabledMileageRates(): should retutn enabled mileage rates', () => {
-    const result = mileageRatesService.filterEnabledMileageRates(mileageRatesData);
+    const result = mileageRatesService.filterEnabledMileageRates(unfilterdMileageRatesData);
```
Comment (Dimple16) [inline src/app/core/services/mileage-rates.service.spec.ts:97]: `unfilterdMileageRatesData` -> typo

File: src/app/core/services/mileage-rates.service.spec.ts [hunk 6]
```diff
@@ -20,24 +32,129 @@ describe('MileageRatesService', () => {
...
     expect(result).toEqual(filterEnabledMileageRatesData);
   });
+
+  it('excludeNullRates(): should exclude mileage rates with null rates', () => {
+    const result = mileageRatesService.excludeNullRates(nullRateIncludedData);
+    expect(result.length).toEqual(nullRateExcludedData1.length);
+    expect(result).toEqual(nullRateExcludedData1);
+  });
+
+  it('should get mileage rates', (done) => {
```
Comment (Dimple16) [inline src/app/core/services/mileage-rates.service.spec.ts:108]: `getMileageRates(): should get mileage rates`

File: src/app/core/services/mileage-rates.service.spec.ts [hunk 7]
```diff
@@ -20,24 +32,129 @@ describe('MileageRatesService', () => {
...
+    mileageRatesService.getAllMileageRatesCount().subscribe((res) => {
+      expect(res).toEqual(platformMileageRatesSingleData.data.length);
+      expect(spenderPlatformV1BetaApiService.get).toHaveBeenCalledOnceWith('/mileage_rates', data);
+      done();
+    });
+  });
+
+  it('getAllMileageRates(): should get all mileage rates', (done) => {
+    const spyGetMileageRates = spyOn(mileageRatesService, 'getMileageRates');
+    spyOn(mileageRatesService, 'getAllMileageRatesCount').and.returnValue(of(100));
```
Comment (Dimple16) [inline src/app/core/services/mileage-rates.service.spec.ts:142]: reduce this count value, use 20,

File: src/app/core/services/mileage-rates.service.spec.ts [hunk 8]
```diff
@@ -20,24 +32,129 @@ describe('MileageRatesService', () => {
...
+
+    // simulate first page of 50 mileage rates
+    const testParams1 = { offset: 0, limit: 50 };
```
Comment (Dimple16) [inline src/app/core/services/mileage-rates.service.spec.ts:145]: change this accordingly

**Review by Dimple16 — CHANGES_REQUESTED** [review]
add return type for this service method - `formatMileageRateName()`

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**